services/brig/src/Brig/API/OAuth.hs

```diff
@@ -348,23 +327,23 @@ lookupAndDeleteByOAuthAuthorizationCode code = lookupOAuthAuthorizationCode <* d
 
 insertOAuthRefreshToken :: (MonadClient m) => Word32 -> Word64 -> OAuthRefreshTokenInfo -> m ()
 insertOAuthRefreshToken maxActiveTokens ttl info = do
-  let rid = refreshTokenId info
+  let rid = info.refreshTokenId
   oldTokes <- determineOldestTokensToBeDeleted <$> lookupOAuthRefreshTokens info.userId
   for_ oldTokes deleteOAuthRefreshToken
-  retry x5 . write qInsertId $ params LocalQuorum (info.userId, rid)
-  retry x5 . write qInsertInfo $ params LocalQuorum (rid, info.clientId, info.userId, C.Set (Set.toList (unOAuthScopes (scopes info))), info.createdAt)
+  retry x5 . write qInsertId $ params LocalQuorum (info.userId, rid, fromIntegral ttl)
+  retry x5 . write qInsertInfo $ params LocalQuorum (rid, info.clientId, info.userId, C.Set (Set.toList (unOAuthScopes info.scopes)), info.createdAt, fromIntegral ttl)
   where
-    qInsertInfo :: PrepQuery W (OAuthRefreshTokenId, OAuthClientId, UserId, C.Set OAuthScope, UTCTime) ()
-    qInsertInfo = fromString $ "INSERT INTO oauth_refresh_token (id, client, user, scope, created_at) VALUES (?, ?, ?, ?, ?) USING TTL " <> show ttl
+    qInsertInfo :: PrepQuery W (OAuthRefreshTokenId, OAuthClientId, UserId, C.Set OAuthScope, UTCTime, Int32) ()
+    qInsertInfo = fromString $ "INSERT INTO oauth_refresh_token (id, client, user, scope, created_at) VALUES (?, ?, ?, ?, ?) USING TTL ?"
 
-    qInsertId :: PrepQuery W (UserId, OAuthRefreshTokenId) ()
-    qInsertId = fromString $ "INSERT INTO oauth_user_refresh_token (user, token_id) VALUES (?, ?) USING TTL " <> show ttl
+    qInsertId :: PrepQuery W (UserId, OAuthRefreshTokenId, Int32) ()
+    qInsertId = fromString $ "INSERT INTO oauth_user_refresh_token (user, token_id) VALUES (?, ?) USING TTL ?"
 
     determineOldestTokensToBeDeleted :: [OAuthRefreshTokenInfo] -> [OAuthRefreshTokenInfo]
     determineOldestTokensToBeDeleted tokens =
       take (length sorted - fromIntegral maxActiveTokens + 1) sorted
       where
-        sorted = sortOn createdAt tokens
+        sorted = sortOn (.createdAt) tokens
```

Comment:
I would buy the record syntax everywhere else, but here it seems redundant. It's a matter of taste, for sure.

Reply:
Yes, I agree. It would be needed if this was an overloaded field. But it is not, so I'll remove it.